core/db.py

```
@@ -144,4 +306,24 @@ def assert_called_once(self) -> None:
+    if _ASYNC_ENGINE is not None:
+        raise RuntimeError(
+            "init_db() cannot be used when async engine is enabled. Use `await init_db_async()` instead."
+        )
+
     metadata.create_all(bind=_RAW_ENGINE)
+
+
+async def init_db_async() -> None:
+    """Async variant of :func:`init_db` for async engines."""
+
+    import core.models  # noqa: F401  # pylint: disable=unused-import
```

Ruff (0.13.2)

Unused `noqa` directive (non-enabled: `F401`)

Remove unused `noqa` directive

(RUF100)